### PR TITLE
Fix Advanced Editor flyouts on mobile with keystone

### PR DESCRIPTION
**Pull request by charrondev**
Closes https://github.com/vanilla/vanilla/issues/8465

This whole legacy flyout situation is probably the biggest definition of spaghetti code that I've ever seen. I'm very glad we don't use it anymore.

Heres what I ended up doing:

- Improve the selectors of the keystone style flyout handling to close more things. Advanced Editor and core fight over settings handlers a lot, so I'm sorry if it's kind of ugly.
- A kludge for the WYSIWYG editor was covering too many selectors and causing flyouts to close and open again immediately. I narrowed the scope of that event handler.

## Testing

I've manually tested various permutations of the following:

- **Themes:** Keystone, Deflector, Basic Mobile
- **Editors:** Markdown, BBCode, WYSIWYG
- Desktop & Mobile devices.

### CODE REVIEW

**Comment by michaeltyson** [comment]
This fixes the Advanced Editor but seems to break at least the "Edit Profile" flyout on a profile page. The menu appears on tap, but disappears when trying to select an item (like Ban - guess it's spammers' lucky day!)

**Comment by charrondev** [comment]
@michaeltyson I think I've got it fixed. I was a bit overzealous with the event handler which affected this page.

**Comment by michaeltyson** [comment]
Yep! That's got it - cheers, @charrondev

**Comment by michaeltyson** [comment]
Ah! It's also broken the top nav menu on mobile. Tapping the menu items now shows nothing (see it at forum.audiob.us)

**Comment by michaeltyson** [comment]
Ah - it's broken the flout from the mobile nav bar. It can't be dismissed any more (see it at forum.audiob.us)

**Comment by charrondev** [comment]
Got it again and double checked all of the interactions. I really wish we could just use our `<DropDown />` component but that would be a hell of a re-factoring.

File: js/flyouts.js [hunk 1]
```diff
@@ -263,6 +272,7 @@
      * Close all of the flyouts unless we are clicking on a button inside of a flyout.
      */
     function handleToggleFlyoutMouseUp() {
+        console.log("flyout mouseup");
```
Comment (michaeltyson) [inline js/flyouts.js]: Was that meant to stay in?

Comment (charrondev) [reply js/flyouts.js]: Good catch. We don't have our linter setup on these files either, which adds to the pain.

It's a bit tough, being used to work in unit tested, CIed coded and coming into these.

**Comment by michaeltyson** [comment]
Yep, that seems to have fixed it! Not sure if it's related, but on the discussion pages, the flyout menu doesn't appear at all (https://open.vanillaforums.com/discussion/37087/profile-menu-on-keystone-on-mobile-hidden-on-discussion-pages#latest). This hasn't changed with this PR, though.

**Pull request merged**